### DISCUSSION TITLE
Plugin system

### DISCUSSION THREAD

**Discussion by MyWay** [stoatchat/discussions#308] [Feature Suggestions]
It would be nice to have a plugin system, so that everybody could take advantage of hooks or whatever it is, to add things to the client. E.g. let's say I want to preload a user avatar, or to add a button in the menu to make a process easier, or to add a reaction to messages, or a filter system for messages and so on. It would also be good because owners could choose which ones to load, instead of having everything preloaded.

**Comment by cyrneko** [comment]
well, I'd take the BetterDiscord approach on this one:

1. Make guidelines describing clearly what is allowed.
2. make everything client-side >>> server-side plugins would then only be on self-hosted instances as it's probably a hassle to neatly integrate them into the main instance of Revolt without having a set of plugins pre-selectable

additionally, one would need to make sure they're crossplatform, that isn't something we should forget
_or_ alternatively add flags to specify what plugins work on what platform to avoid confusion

though, overall, I like the idea, was actually one of the first things I hoped for Revolt to get once I discovered it recently :)
